openstack_controller/datadog_checks/openstack_controller/api/api_rest.py

```diff
@@ -437,6 +453,39 @@ def get_baremetal_conductors(self):
 
         return self.make_paginated_request(url, 'conductors', 'hostname', params={})
 
+    def get_baremetal_volume_connectors(self):
+        return self.make_paginated_request(
+            '{}/v1/volume/connectors'.format(self._catalog.get_endpoint_by_type(Component.Types.BAREMETAL.value)),
+            'connectors',
+            'uuid',
+        )
+
+    def get_baremetal_volume_targets(self):
+        return self.make_paginated_request(
+            '{}/v1/volume/targets'.format(self._catalog.get_endpoint_by_type(Component.Types.BAREMETAL.value)),
+            'targets',
+            'uuid',
+        )
+
+    def get_baremetal_drivers(self):
+        response = self.http.get(
+            '{}/v1/drivers'.format(self._catalog.get_endpoint_by_type(Component.Types.BAREMETAL.value))
+        )
+        response.raise_for_status()
+        return response.json().get('drivers', [])
+
+    def get_baremetal_allocations(self):
+        if float(self.config.ironic_microversion) < 1.52:
+            self.log.info(
+                "Ironic microversion is below 1.52 and set to %s, cannot collect allocations",
```

Comment:
Can you add a test when the ironic microversion is low and you do not see allocation metrics but do see this log line?

Reply:
Added the test (test_allocations_low_microversion)